crates/p2p_protocol/protocol/transit_encoding.capnp

```diff
@@ -0,0 +1,153 @@
+# lib3h In-Transit Encoding Handshake Protocol
+# These are all separate structs, rather than having a top-level root/union
+# This works because there is a well-defined handshake expectation
+# Anything unexpected is an error which should cause a connection drop
+
+@0xdf3e1d96ae657243;
+
+struct Halt {
+  # Let's at least try to be nice to people connecting to us
+  # Give them some info why we won't accept their connection
+
+  reasonCode @0 :ReasonCode;
+  reasonText @1 :Text;
+
+  enum ReasonCode {
+    unspecified @0;
+    badMagic @1;
+    badEncoding @2;
+    badNetworkId @3;
+    kxDecodeFail @4;
+    unexpectedSigPubKey @5;
+    badSignature @6;
+    encDecodeFail @7;
```

Comment:
Assuming it's easy to expand reason codes in the future.  But for the moment makes things brittle?  Would it make sense to allow text into the reason?  Or something app-dependant, like "banned" or "blocked by warrant< hash>".

Reply:
Yep! there's the reasonText above, so if the error isn't a well-known code, the code can be unspecified and the error can be enumerated in text.

We may want to remove some of these codes... really, well-known codes are only useful if they are something that can be recovered from in an automated code path. Like the `WouldBlock` error sent back from a non-blocking tcp session that means, we don't have data right now, try back later.